src/materials/tdymaterialproperties.c

```diff
@@ -443,289 +530,233 @@ static PetscErrorCode FullTensorWrapperFunction(void *context, PetscInt n,
   PetscFunctionReturn(0);
 }
 
-/// Sets the porosity function to the given constant value.
-PetscErrorCode MaterialPropSetConstantPorosity(MaterialProp *matprop,
-                                               PetscReal porosity) {
-  PetscErrorCode ierr;
-  PetscFunctionBegin;
-
-  void *context;
-  PetscReal value[9];
-  value[0] = porosity;
-  ierr = CreateConstantContext(matprop->dim, value, &context); CHKERRQ(ierr);
-  ierr = MaterialPropSetPorosity(matprop, context, ConstantScalarWrapperFunction,
-                                 TDyFree); CHKERRQ(ierr);
-  PetscFunctionReturn(0);
+// These macros are used to support the various material model quantities.
```

Comment:
This is nice. I never think about using macros and this provides a concrete example of when one should consider using a macro.